algorithmic_efficiency/data_utils.py

```diff
@@ -23,14 +23,14 @@ def shard_and_maybe_pad_np(
   create the corresponding mask, and reshape it to be sharded across devices.
   """
   local_device_count = max(torch.cuda.device_count(), jax.local_device_count())
-  current_batch_size = batch['inputs'].shape[0]
+  current_batch_size = batch['inputs'][0].shape[0]
```

Comment:
Ah right, librispeech has a tuple in `batch['inputs']`. This should be something like
```python
inputs = batch['inputs']
current_batch_size = inputs[0].shape[0] if isinstance(inputs, tuple) else inputs.shape[0]
```

Reply:
+1 we need to make sure this works for all workloads

Reply:
made the changes

Reply:
I think I tested for all workloads, but librispeech and Criteo1TB in the original PR (didn't have the data ready).

Reply:
I think I tested this for all workloads, but librispeech and Criteo1TB in the original PR (didn't have the data ready).